bin/lms-rtparser.php

```diff
@@ -670,7 +673,45 @@ public function doValidate(&$uri, $config, $context)
             }
         }
 
-        if (!$prev_tid) { // generate new ticket if previous not found
+        // generate new messege if previous ticket found or ticket was resolved
+        // less than rt.allow_modify_resolved_tickets_newer_than variable value seconds
+
+        if (($prev_tid && $prev_tid_contents['state'] != RT_RESOLVED) ||
+            ($prev_tid && $prev_tid_contents['state'] == RT_RESOLVED && ($prev_tid_contents['resolvetime'] + $modify_ticket_timeframe) > time())) {
```

Comment:
W obydwu częściach || występuje $prev_tid. Lepiej wyciągnąć to na początek jak warunek nadrzędny.

Reply:
Można jeszcze bardziej zoptymalizować - niepotrzebnie tyle testów stanu zgłoszenia.
Docelowo mogłoby być chyba tak:
```php
if ($prev_tid && ($prev_tid_contents['state'] != RT_RESOLVED || $prev_tid_contents['resolvetime'] + $modify_ticket_timeframe > time())) {
```

Reply:
Optymalizacja i uproszczanie kodu zawsze w cenie.